cylc/flow/scripts/workflow_state.py

```diff
@@ -18,252 +18,260 @@
+        try:
+            self.workflow_id = infer_latest_run_from_id(
+                self.workflow_id_raw,
+                self.alt_cylc_run_dir
+            )
+        except InputError:
+            LOG.debug("Workflow not found")
+            return False
+
+        # Print inferred workflow ID.
+        sys.stderr.write(f"{self.workflow_id}\n")
+
+        return True
+
+    def _db_connect(self) -> bool:
+        """Connect to workflow DB, return True if connected."""
+        try:
+            self.db_checker = CylcWorkflowDBChecker(
+                get_cylc_run_dir(self.alt_cylc_run_dir),
+                self.workflow_id
+            )
+        except (OSError, sqlite3.Error):
+            LOG.debug("DB not connected")
+            return False
+
+        return True
+
+    async def check(self) -> bool:
+        """Return True if requested state achieved, else False.
+
+        Called once per poll by super() so only find and connect once.
+
+        Store self.result for external access.
+
+        """
+        if self.workflow_id is None and not self._find_workflow():
+            return False
+
+        if self.db_checker is None and not self._db_connect():
+            return False
+
+        if self.cycle is None:
+            # Adjust target cycle point to the DB format.
+            self.cycle = self.db_checker.adjust_point_to_db(
+                self.cycle_raw, self.offset)
+
+        self.result = self.db_checker.workflow_state_query(
+            self.task, self.cycle, self.task_sel, self.is_output, self.flow_num
+        )
+        if self.result:
+            # End the polling dot stream and print inferred runN workflow ID.
+            self.db_checker.display_maps(self.result, self.old_format)
+
+        return bool(self.result)
 
 
 def get_option_parser() -> COP:
     parser = COP(
         __doc__,
-        argdoc=[WORKFLOW_ID_ARG_DOC]
+        argdoc=[ID_MULTI_ARG_DOC]
     )
 
     parser.add_option(
-        "-t", "--task", help="Specify a task to check the state of.",
-        action="store", dest="task", default=None)
-
-    parser.add_option(
-        "-p", "--point",
-        help="Specify the cycle point to check task states for.",
-        action="store", dest="cycle", default=None)
-
-    parser.add_option(
-        "-T", "--task-point",
-        help="Use the CYLC_TASK_CYCLE_POINT environment variable as the "
-             "cycle point to check task states for. "
-             "Shorthand for --point=$CYLC_TASK_CYCLE_POINT",
-        action="store_true", dest="use_task_point", default=False)
-
-    parser.add_option(
-        "-d", "--run-dir",
-        help="The top level cylc run directory if non-standard. The "
-             "database should be DIR/WORKFLOW_ID/log/db. Use to interrogate "
-             "workflows owned by others, etc.; see note above.",
-        metavar="DIR", action="store", dest="alt_run_dir", default=None)
+        "-d", "--alt-cylc-run-dir",
```

Comment:
Similar to my other comments - Is it wise to make this a totally breaking change rather than deprecating the older style of xtriggers?

Reply:
c.f. my comment on the xtrigger - the problem is we need the "--output" option for the trigger name now, not the corresponding task message, and on master `--message` and `--output` both query the task message, which makes back-compat difficult. 

However, now that I have managed to achieve back-compat for the xtrigger, I think I can do it for the CLI as well by considering all the command line as a whole.

Reply:
No longer a breaking change. (I was able to do this, because the old `--output` and `--message` args were only a problem for 8.0-8.2.x DBs, which only store the message).